### ISSUE TITLE
Automated IV Drip only injecting at 5u

### ISSUE THREAD

**Issue by ZeroKelvn** [tgstation/tgstation#71579]
Reporting client version: 514.1588

## Round ID:
195210

## Testmerges:
- [Psykers](https://www.github.com/tgstation/tgstation/pull/71566)
- [fixes & increases SSinput click delay threshold & moves it and average_click_delay to deciseconds](https://www.github.com/tgstation/tgstation/pull/71520)
- [Revert "Unit tests radio, saymode, and language prefix keys"](https://www.github.com/tgstation/tgstation/pull/71474)
- [Partially reverts moving up/down whilst in a movable from #70431](https://www.github.com/tgstation/tgstation/pull/71527)

## Reproduction: An automated IV drip from plumbing only injected the 5u dose no matter what the value was set to, even at 0.
1. Obtain a Recurring Rainbow slime extract (this is easiest to display since it has a different reaction at 5u Plasma)
2. Hook up an Automated IV drip to draw from a supply of liquid plasma
3. Connect the IV drip to the extract
4. Watch the slime core expand uncontrollably despite the injection rate still being at 0
This seems to be a bug coming from https://github.com/tgstation/tgstation/pull/71217#issue-1446127798, which allowed this sort of interaction to happen in the first place. I'm not sure if I was doing something wrong, but given how niche plumbing seems to be in the first place, it's probably best to pop in on here to see if anyone can reproduce it or figure what was going on.

**Comment by Time-Green** [comment]
I'll look into it
@MTandi you have any idea?

**Comment by MTandi** [comment]
Yeah, the plumbing IV drip seems to be ignoring transfer rate and uses the rate of plumbing system. I.e. if you make the chem synthesizer generate 1 unit instead of 5, it will transfer with 1. I didn't notice this when I was testing automated drip. 

Not sure how to resolve this though, need to dig into plumbing.

And my PR is unrelated. It seems like automated drips always ignored transfer rate if the current code doesn't work.

**Comment by Time-Green** [comment]
Eh plumbing is my domain anyway, I’ll put it on mu TODO

**Comment by Time-Green** [comment]
Yeah okay, iv only supports having one container and to not mix up incoming and outcoming reagents, receiving reagents are dumped straight into the target object by the plumbing component. The actual fix would require a bit of a rework of IV drips themselves to make them support multiple containers

**Comment by MTandi** [comment]
I can replace the transfer rate controls with a text saying "Injection rate controlled by the network" for plumbing drip. It works fine for blood draining though, because it goes into the internal container.

**Comment by Time-Green** [comment]
> I can replace the transfer rate controls with a text saying "Injection rate controlled by the network" for plumbing drip. It works fine for blood draining though, because it goes into the internal container.

Works for me, but I'll keep this issue open

**Comment by ZeroKelvn** [comment]
> Yeah okay, iv only supports having one container and to not mix up incoming and outcoming reagents, receiving reagents are dumped straight into the target object by the plumbing component. The actual fix would require a bit of a rework of IV drips themselves to make them support multiple containers

I've been thinking about this for a while, and I'm not really sure I understand where the big part of the issue is? Please take this with a grain of salt, as I hardly understand the code in this game. As I understand it though, the automated IV drip injects at a maximum rate of 5u based on the contents of its receiving end, without ever storing them inside the IV. Conversely, reagents drained from the target are held in its inventory, and discharged to the output. I don't see how the reagents would ever mix in this case, even if the input was set below 5u. I would think that modifying the injection rate from the automated IV drip could also moderate how many reagents it was pulling in, almost like a reaction chamber set to a low pull dose, and still dump them straight into the target. This would bypass the need for a second reagent container, since the automated IV drip would still only hold the drawn chemicals.

**Comment by Time-Green** [comment]
> > Yeah okay, iv only supports having one container and to not mix up incoming and outcoming reagents, receiving reagents are dumped straight into the target object by the plumbing component. The actual fix would require a bit of a rework of IV drips themselves to make them support multiple containers
> 
> I've been thinking about this for a while, and I'm not really sure I understand where the big part of the issue is? Please take this with a grain of salt, as I hardly understand the code in this game. As I understand it though, the automated IV drip injects at a maximum rate of 5u based on the contents of its receiving end, without ever storing them inside the IV. Conversely, reagents drained from the target are held in its inventory, and discharged to the output. I don't see how the reagents would ever mix in this case, even if the input was set below 5u. I would think that modifying the injection rate from the automated IV drip could also moderate how many reagents it was pulling in, almost like a reaction chamber set to a low pull dose, and still dump them straight into the target. This would bypass the need for a second reagent container, since the automated IV drip would still only hold the drawn chemicals.

Yes but would require a lot of shitcode or a proper big rework since the plumbing part and the IV part are seperate

**Comment by MTandi** [comment]
I honestly thought the same way as @ZeroKelvn. If I would make an plumbing IV drip from scratch, I would make it without any containers. The container is the attached object and the transfer rates only modifies the plumbing supply/demand. Containers act as a buffer and are optional qol features. Right now it is weird that the buffer exists for the output, but doesn't exist for the input.

Though I need to dig into plumbing code to understand whether it is simple to modify the supply/demand of the plumbing node and make it work. Didn't bother.
